### ISSUE TITLE
Connection to existing cluster (created with kubeadm)

### ISSUE THREAD

**Issue by dkirrane** [Azure/vscode-kubernetes-tools#63]
When I try the Kubernetes Create command I hit the below error. So I assume I need to connect to a cluster first? Is there a way to configure the extension to use a kubeadm cluster?

![image](https://user-images.githubusercontent.com/1280116/36676399-7ea7381c-1b03-11e8-8888-b91dfbce28ba.png)

**Comment by itowlson** [comment]
Correct.  The extension uses your active `kubectl` context (in other words, the cluster to which `kubectl` is currently connected).  To activate a particular cluster you could either:

* Open the Kubernetes tree view in the explorer pane, select your kubeadm cluster, right-click and choose "Set as Current Context"
* In the Terminal, run `kubectl config use-context <your_context_name>`

If your cluster isn't currently in your kubeconfig file, then you'll need to add it.  The extension doesn't currently provide a way to do this; you'll need to use cluster-specific tools or `kubectl config` sub-commands.

**Comment by dkirrane** [comment]
Hi,

So I've connected to my cluster in the terminal
![image](https://user-images.githubusercontent.com/1280116/36819211-5e95a26e-1ce0-11e8-9f96-ad2fc34fad42.png)

But my cluster doesn't appear in the explorer
![image](https://user-images.githubusercontent.com/1280116/36819279-c3751a0c-1ce0-11e8-9069-e346dd2a33ca.png)

And I can only configure Azure clusters
![image](https://user-images.githubusercontent.com/1280116/36819316-f5aa8db8-1ce0-11e8-9a1c-d0f2b7881cf2.png)

Does vs-kubernetes expect the kubeconfig to be in a certain folder? For instance mine resides in `%HOMEPATH%\.kube\dev\config`

**Comment by dkirrane** [comment]
Does the extension expect the kubernetes config file in specific location. Mine is under `%USERHOME%\.kube\dev\config`

Connected to my kubeadm k8s cluster now usin terminal
![image](https://user-images.githubusercontent.com/1280116/36819503-aa6c2266-1ce1-11e8-960e-d36afcdd1094.png)

Nothing appears in explorer config k8s pane
![image](https://user-images.githubusercontent.com/1280116/36819487-9c31ed7a-1ce1-11e8-92c6-0eef1f68d7b4.png)

And only Azure options in 
![image](https://user-images.githubusercontent.com/1280116/36819516-b4a69824-1ce1-11e8-8eaf-f7f7d4ea54aa.png)

**Comment by itowlson** [comment]
We don't expect a particular kubeconfig location, we just invoke kubectl.  But clearly we're not getting any output from kubectl in your case.  Your kubeconfig does look like a non-default location.  Does your terminal have a KUBECONFIG environment variable set up?  If so, is that same environment variable set up in the VS Code process?  If kubectl is using a different (and empty) config file when we call it, that would explain the behaviour you're seeing I think...

Re the Configure pane - yeah, we only have configure from Azure built in at the moment.  Would love to talk about how we can support configuring from a kubeadm cluster too.  But if your kubectl is working then you shouldn't need to do the configure step.

Thanks for your patience in investigating this!

**Comment by bnookala** [comment]
@dkirrane would you be able to run an `echo $env:KUBECONFIG` in a powershell window? We're considering making the kubeconfig location configurable in the extension, but need to confirm why the extension isn't picking up on the location of your kubeconfig.

**Comment by squillace** [comment]
@itowlson: based on feedback in Beijing and several sporadic reports, we should be able to load a config visually along with "Add Existing Cluster". Can we promote this to a feature request, and review whether it's required?

**Comment by itowlson** [comment]
@squillace Yes, absolutely.  Can you expand on what you mean by "load a config visually"?  Do you mean having a command (or right-click option) to say "I want to use a nondefault kubeconfig file" and have it prompt you for the kubeconfig?  What is the scope, e.g. does the selected config remain in force from session to session, or only for the current session?

**Comment by brendandburns** [comment]
As a temporray workaround, note you can merge multiple kubeconfigs with (slightly hacky)

```
KUBECONFIG=`/path1/config:/path2/config`

And then run `kubectl config view`

that will give you the merged kubeconfig that you can place in a standard location.

But I will also implement this for real.
